pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -158,6 +148,37 @@ func ValidateClusterConfig(cfg *ClusterConfig) error {
 	return nil
 }
 
+func validateCloudWatchLogging(clusterConfig *ClusterConfig) error {
+	if !clusterConfig.HasClusterCloudWatchLogging() {
+		if clusterConfig.CloudWatch.ClusterLogging.LogRetentionInDays != 0 {
+			return errors.New("cannot set cloudWatch.clusterLogging.logRetentionInDays without enabling log types")
+		}
+		return nil
+	}
+
+	for i, logType := range clusterConfig.CloudWatch.ClusterLogging.EnableTypes {
+		isUnknown := true
+		for _, knownLogType := range SupportedCloudWatchClusterLogTypes() {
+			if logType == knownLogType {
+				isUnknown = false
+			}
+		}
+		if isUnknown {
+			return errors.Errorf("log type %q (cloudWatch.clusterLogging.enableTypes[%d]) is unknown", logType, i)
+		}
+	}
```

Comment:
this bit is missing from the unit tests

Reply:
It's already covered here: https://github.com/weaveworks/eksctl/blob/b9dd6e0e4c4f505c5595a074e13d2cf801ef198e/pkg/apis/eksctl.io/v1alpha5/validation_test.go#L460

Reply:
ah yes, great! :) didn't see `It("should handle unknown types", func() {` because it wasn't in the diff.